pkg/dbal/DbalConsumerHelperTrait.php

```diff
@@ -91,16 +101,28 @@ protected function redeliverMessages(): void
             ->setParameter('redelivered', true, Type::BOOLEAN)
             ->execute()
         ;
+
+        $this->redeliverMessagesLastExecutedAt = microtime(true);
     }
 
     protected function removeExpiredMessages(): void
     {
+        if (null === $this->removeExpiredMessagesLastExecutedAt) {
+            $this->removeExpiredMessagesLastExecutedAt = microtime(true);
+        }
+
+        if ((microtime(true) - $this->removeExpiredMessagesLastExecutedAt) < 1) {
```

Comment:
you could put that in an `elseif` so it's not executed needlessly on the first execution

Reply:
or just put `null === $this->removeExpiredMessagesLastExecutedAt || ....` and remove the above condition